File: app/src/main/java/com/sjsoft/app/constant/AppConfig.java
package com.sjsoft.app.constant;

import com.sjsoft.app.BuildConfig;

public class AppConfig {
    public static class AppMode{
        public static final int production = 0;
        public static final int PGReview = 1;
    }

    //public static long cacheTime_nationalValue = TimeUnit.MINUTES.toMillis(1);

    public final static String prod = "prod";
    public final static String dev = "dev";
    public final static String stg = "stg";
    public final static String mock = "mock";

    public static int getAppMode() {
        return AppMode.production;
    }

    public static boolean isProductionVersion() {
        return BuildConfig.FLAVOR.contains(prod) && !BuildConfig.DEBUG;
    }

    public static boolean isDevVersion() {
        return BuildConfig.FLAVOR.contains(dev);
    }

    public static boolean isStgVersion() {
        return  BuildConfig.FLAVOR.contains(stg);
    }



    public static boolean isMockVersion(){
        return BuildConfig.FLAVOR.contains(mock);
    }
}

File: app/src/main/java/com/sjsoft/app/constant/AppConfig.kt
package com.sjsoft.app.constant

import com.sjsoft.app.BuildConfig

class AppConfig {
    companion object {
        const val prod = "prod"
        const val dev = "dev"
        const val stg = "stg"
        const val mock = "mock"

        fun needDebugInfo(): Boolean {
            return !isProductionVersion || BuildConfig.DEBUG
        }

        val isProductionVersion: Boolean
            get() = BuildConfig.FLAVOR.contains(prod) && !BuildConfig.DEBUG

        val isDevVersion: Boolean
            get() = BuildConfig.FLAVOR.contains(dev)

        val isStgVersion: Boolean
            get() = BuildConfig.FLAVOR.contains(stg)

        val isMockVersion: Boolean
            get() = BuildConfig.FLAVOR.contains(mock)


        const val splashDelay = 3000L
        const val enteringDelay = 400L

        //Pixlee
        const val albumId = "5984962"
        const val LIST_VISIBLE_THRESHOLD = 5
        const val pixleeEmail = "sdk@pixleeteam.com"
        const val pixleeUserName = "SDK Project"

        fun getS3Url(key: String): String {
            return "${BuildConfig.AWS_S3_DOMAIN}$key"
        }

    }

    object DeepLinkParams {
        const val drwNo = "drwNo"
    }
}